eZ/Bundle/EzPublishCoreBundle/Command/RegenerateUrlAliasesCommand.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\Command;
+
+use eZ\Publish\API\Repository\LocationService;
+use eZ\Publish\API\Repository\PermissionResolver;
+use eZ\Publish\API\Repository\SearchService;
+use eZ\Publish\API\Repository\URLAliasService;
+use eZ\Publish\API\Repository\UserService;
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\Subtree;
+use eZ\Publish\Core\Pagination\Pagerfanta\LocationSearchAdapter;
+use Pagerfanta\Pagerfanta;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * The ezplatform:urls:regenerate-aliases Symfony command implementation.
+ * Recreates system URL aliases for all existing Locations and cleanups corrupted URL alias nodes.
+ */
+class RegenerateUrlAliasesCommand extends Command
+{
+    /**
+     * @var \eZ\Publish\API\Repository\PermissionResolver
+     */
+    private $permissionResolver;
+
+    /**
+     * @var \eZ\Publish\API\Repository\UserService
+     */
+    private $userService;
+
+    /**
+     * @var \eZ\Publish\API\Repository\SearchService
+     */
+    private $searchService;
+
+    /**
+     * @var \eZ\Publish\API\Repository\LocationService
+     */
+    private $locationService;
+
+    /**
+     * @var \eZ\Publish\API\Repository\URLAliasService
+     */
+    private $urlAliasService;
+
+    public function __construct(
+        PermissionResolver $permissionResolver,
+        UserService $userService,
+        SearchService $searchService,
+        LocationService $locationService,
+        URLAliasService $urlAliasService
+    ) {
+        parent::__construct(null);
+
+        $this->permissionResolver = $permissionResolver;
+        $this->userService = $userService;
+        $this->searchService = $searchService;
+        $this->locationService = $locationService;
+        $this->urlAliasService = $urlAliasService;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('ezplatform:urls:regenerate-aliases')
+            ->setDescription(
+                'Regenerates Location URL aliases (autogenerated) and migrates custom Location ' .
+                'and global URL aliases with Legacy Storage Engine'
+            )
+            ->addOption(
+                'bulk-count',
+                'c',
+                InputOption::VALUE_OPTIONAL,
+                'Number of Locations fetched into memory and processed at once',
+                50
```

Comment:
Perhaps it might be a constant to be consistent with:
https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Bundle/EzPublishCoreBundle/Command/CheckURLsCommand.php#L22 &
https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Bundle/EzPublishCoreBundle/Command/ResizeOriginalImagesCommand.php#L41

Reply:
Nice, let's be consistent. Fixed via f96dd03 along with some other fixes.